api/src/main/java/marquez/api/DatasetResource.java

```diff
@@ -150,6 +151,28 @@ public Response list(
     return Response.ok(new ResultsPage<>("datasets", datasets, totalCount)).build();
   }
 
+  @Timed
+  @ResponseMetered
+  @ExceptionMetered
+  @DELETE
+  @Path("{dataset}")
+  @Produces(APPLICATION_JSON)
+  public Response deleteDataset(
+      @PathParam("namespace") NamespaceName namespaceName,
+      @PathParam("dataset") DatasetName datasetName) {
+    throwIfNotExists(namespaceName);
+
+    Dataset dataset =
```

Comment:
Not sure why do we need this. Wouldn't `datasetService.delete` be enough? We don't have to return a deleted dataset in a body.

Reply:
There is pattern in Marquez where it returns object on which the operation was carried out. I just decided to follow it.